server.go

```diff
@@ -605,6 +605,18 @@ func (r *oauthProxy) newOpenIDClient() (*oidc.Client, oidc.ProviderConfig, *http
 	// step: create a idp http client
 	hc := &http.Client{
 		Transport: &http.Transport{
+			Proxy: func(_ *http.Request) (*url.URL, error) {
+				if r.config.OpenIDProviderProxy != "" {
```

Comment:
Out of interest what's the reason behind using is via an option rather than relying on ```Proxy: http.ProxyFromEnvironment```

Reply:
This implementation is explicit proxy configuration, and it provides more flexibility for configuration: config file, command line option, (env variable).